From 64d0dc7839758cafddcd81e3087b13bb9c337e75 Mon Sep 17 00:00:00 2001
From: eInfochips <eInfochips@eInfochips.com>
Date: Fri, 25 Jan 2019 17:14:55 +0530
Subject: [PATCH] Gateway-DB410C: Support for Factory reset gpio

- SW4 in gateway board is connected with GPIO-69
  and used as factory reset button gpio.
- configure GPIO_69 as input pin and configure
  as interrupt pin.
- Added support to monitor this factory reset
  button in growhouse server.

Test-Done:
----------
- growhouse-server service was running in DB410C
  and pressed Factory reset button for more than 3 sec
  and factory_button_longPress_detected callback was
  called into growhouse-server application.

Note: This test was perform on DB410C board using GPIO107,
      volume up button.

Signed-off-by: eInfochips <eInfochips@eInfochips.com>

diff --git a/arch/arm64/boot/dts/qcom/apq8016-sbc-soc-pins.dtsi b/arch/arm64/boot/dts/qcom/apq8016-sbc-soc-pins.dtsi
index 21d0822..34f127b 100644
--- a/arch/arm64/boot/dts/qcom/apq8016-sbc-soc-pins.dtsi
+++ b/arch/arm64/boot/dts/qcom/apq8016-sbc-soc-pins.dtsi
@@ -86,4 +86,16 @@
 			bias-pull-up;
 		};
 	};
+
+	factory_reset_button: factory_reset_button {
+		pinmux {
+			function = "gpio";
+			pins = "gpio69";
+		};
+		pinconf {
+			pins = "gpio69";
+			drive-strength = <16>;
+			bias-disable;
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi b/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
index 862b4e0..6f861bf 100644
--- a/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
+++ b/arch/arm64/boot/dts/qcom/apq8016-sbc.dtsi
@@ -481,12 +481,12 @@
 		autorepeat;
 
 		pinctrl-names = "default";
-		pinctrl-0 = <&msm_key_volp_n_default>;
+		pinctrl-0 = <&factory_reset_button>;
 
 		button@0 {
-			label = "Volume Up";
-			linux,code = <KEY_VOLUMEUP>;
-			gpios = <&msmgpio 107 GPIO_ACTIVE_LOW>;
+			label = "FACTORY_RESET_BUTTON";
+			linux,code = <BTN_0>;
+			gpios = <&msmgpio 69 GPIO_ACTIVE_LOW>;
 		};
 	};
 };
-- 
2.7.4

